frontend/src/components/StatusBadge.jsx

```diff
@@ -0,0 +1,34 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+
+export default function StatusBadge({ status }) {
+  let colorClass = "text-bg-dark";
+  let label = status;
+
+  // Don't render anything if the status is null
+  if (!status) {
+    return null;
+  }
+
+  // Map status code to color class and display label
+  const statusMap = new Map([
+    ["published", { cssClass: "text-bg-primary", displayText: "Published" }],
+    ["approved", { cssClass: "text-bg-success", displayText: "Approved" }],
+    ["requested", { cssClass: "text-bg-warning", displayText: "Requested" }],
+  ]);
+
+  if (statusMap.has(status)) {
+    const { cssClass, displayText } = statusMap.get(status);
+
+    colorClass = cssClass;
+    label = displayText;
+  }
+
+  const classes = classNames(["badge", "rounded-pill", colorClass]);
+
+  return <span className={classes}>{label}</span>;
+}
+
+StatusBadge.propTypes = {
+  status: PropTypes.string,
+};
```

Comment:
Isn't the park status supposed to be required? or could it be without status?

Reply:
I asked for clarification on Slack because I'm actually not sure if there are any cases where it could be blank